app/Services/Model/Imp/SummitService.php

```diff
@@ -3571,4 +3580,23 @@ public function deleteEventFeedback(Summit $summit, int $event_id, int $feedback
 
         });
     }
+
+    /**
+     * @param Summit $summit
+     * @return string
+     * @throws ValidationException
+     */
+    public function generateQREncKey(Summit $summit):string
+    {
+        return $this->tx_service->transaction(function () use ($summit) {
+
+            if (!is_null($summit->getQRCodesEncKey()))
+                throw new ValidationException("There is already a QR encryption key configured for this Summit");
+
+            $enc_key = $this->encryption_key_generator->generate();
```

Comment:
how do u ensure that the key is unique per summit

Reply:
done